src/main/java/hudson/plugins/jira/JiraSite.java

```diff
@@ -811,7 +809,7 @@ public String getPropertyValue(final String s) {
      */
     @Nullable
     public URL getUrl() {
-        return Objects.firstNonNull(this.url, this.alternativeUrl);
+        return this.url != null ? this.url : this.alternativeUrl;
```

Comment:
We had multiple misinterpretations of the usage of main url and alternativeUrl, when touching this I'm wondering if we couldn't already just use `this.url` here and treat alternativeUrl as a totally separate thing used for generated links/views only (not for API communication). In that case, this method should be actually `@NotNull﻿`, if not, then I guess `CheckForNull` instead of `Nullable` per [this discussion](https://intellij-support.jetbrains.com/hc/en-us/community/posts/115000214944-Why-javax-annotation-CheckForNull-is-not-included-in-nullable-annotations-set-by-default-)

Reply:
sounds like too much change? :)

Reply:
Sure, I agree, just wanted to bring this to our discussion :)